core/base/src/TString.cxx

```diff
@@ -1450,91 +1449,83 @@ TString operator+(const TString &s, char c)
 }
 
 ////////////////////////////////////////////////////////////////////////////////
-/// Add integer to string.
+/// Add string to char.
 
-TString operator+(const TString &s, Long_t i)
-{
-   char si[32];
-   snprintf(si, sizeof(si), "%ld", i);
-   return TString(s.Data(), s.Length(), si, strlen(si));
-}
-
-////////////////////////////////////////////////////////////////////////////////
-/// Add integer to string.
-
-TString operator+(const TString &s, ULong_t i)
-{
-   char si[32];
-   snprintf(si, sizeof(si), "%lu", i);
-   return TString(s.Data(), s.Length(), si, strlen(si));
-}
-
-////////////////////////////////////////////////////////////////////////////////
-/// Add integer to string.
-
-TString operator+(const TString &s, Long64_t i)
+TString operator+(char c, const TString &s)
 {
-   char si[32];
-   snprintf(si, sizeof(si), "%lld", i);
-   return TString(s.Data(), s.Length(), si, strlen(si));
+   return TString(&c, 1, s.Data(), s.Length());
 }
 
 ////////////////////////////////////////////////////////////////////////////////
-/// Add integer to string.
+/// Add signed integer to string.
 
-TString operator+(const TString &s, ULong64_t i)
+template<class T>
+typename std::enable_if<is_signed_numeral<T>::value,TString>::type
+operator+(const TString &s, T i)
```

Comment:
The template definition must be in the header (and even right at the `friend` declaration if I'm not mistaken).